src/cache/cache_las.c

```diff
@@ -1423,10 +1434,50 @@ __wt_find_lookaside_upd(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDAT
         upd->durable_ts = durable_timestamp;
         upd->start_ts = las_timestamp;
         upd->prepare_state = prepare_state;
-        /*
-         * Mark this update as external and to be discarded when not needed.
-         */
-        upd->ext = 1;
+
+        if (prepare_state == WT_PREPARE_INPROGRESS) {
+            WT_ASSERT(session, !modify);
+            WT_ASSERT(session, cbt->compare == 0);
```

Comment:
Can you put a comment why we are asserting `cbt->compare == 0`

Reply:
fd82c8e

I've added a comment and moved the assertion to the top of the function since it actually applies everywhere.